app/routes.py

```diff
@@ -1,2 +1,251 @@
+
+        url_path = "https://slack.com/api/chat.postMessage"
+        SLACK_API_KEY = os.environ.get("SLACK_API_KEY")
+
+        header = {
+            "Authorization": f"Bearer {SLACK_API_KEY}"
+            }
+
+        query_params = {
+            "channel": "task-notifications",
+            "text": f"Someone just completed the task {task.title}"
+        }
+
+        requests.post(url_path, params=query_params, headers=header)
+
+    else: 
+        task.completed_at = None
+
+    db.session.commit()
+    return {"task": task.task_to_json()}, 200
+
+
+#WAVE 5
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"], strict_slashes=False)
+def create_goal():
+    request_body = request.get_json()
+    
+    if "title" not in request_body:
+        return jsonify(details="Invalid data"), 400
+    
+    new_goal = Goal(title=request_body["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+    return {"goal": new_goal.goal_to_json()}, 201
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_saved_goals():
+    
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append(goal.goal_to_json())
+
+    return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"], strict_slashes=False)
+def get_single_goal(goal_id):
+
+    if not is_int(goal_id):
+        return {
+            "message": f"ID {goal_id} must be an integer",
+            "success": False
+        }, 400
+
+    goal = Goal.query.get(goal_id)
+    if goal:
+        return {"goal": goal.goal_to_json()}, 200
+    else:
+        return Response("",status=404)
+    
+@goals_bp.route("/<goal_id>", methods=["PUT"], strict_slashes=False)
+def update_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    if goal == None:
+        return Response("", status=404)
+    
+    if goal: 
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+        return {"goal": goal.goal_to_json()}, 200
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"], strict_slashes=False)    
+def delete_single_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal == None:
+        return Response("", status=404)
+
+    if goal:
+        db.session.delete(goal)
+        db.session.commit()
+
+        return {
+            "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"}, 200
+
+#WAVE 6
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_task_into_goal(goal_id):
+
+    if not is_int(goal_id):
+        return {
+            "message": f"ID {goal_id} must be an integer",
+            "success": False
+        }, 400
+
+    goal = Goal.query.get(goal_id)
+    request_body = request.get_json()
+
+    if goal == None:
+        return Response("", status=404)
+
+    for task_id in request_body["task_ids"]:
```

Comment:
Helper variables can improve readability similar to helper functions.  Storing `request_body["task_ids"]` into a variable with a meaningful name, and then using that throughout the rest of your code can make it easier to see what's happening.